### PR TITLE
add parameter out to four mathematical functions

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: dpnp/dpnp_algo/dpnp_algo.pyx [hunk 1]
```diff
@@ -258,19 +258,29 @@ cdef dparray call_fptr_1out(DPNPFuncName fptr_name, dparray_shape_type result_sh
...
-    """ Create result array with type given by FPTR data """
-    cdef dparray result = utils.create_output_array(result_shape, kernel_data.return_type, None)
+    cdef dparray result
+
+    if out is not None:
+        if out.dtype != result_type:
+            utils.checker_throw_value_error(func_name, 'out.dtype', out.dtype, result_type)
+        if out.shape != result_shape:
+            utils.checker_throw_value_error(func_name, 'out.shape', out.shape, result_shape)
+        result = out
```
Comment (samir-nasibli) [inline dpnp/dpnp_algo/dpnp_algo.pyx:278]: see comment for #786

File: dpnp/dpnp_algo/dpnp_algo_mathematical.pyx [hunk 2]
```diff
@@ -339,8 +339,38 @@ cpdef dparray dpnp_negative(dpnp_descriptor x1):
     return call_fptr_1in_1out(DPNP_FN_NEGATIVE, x1, x1.shape)
 
 
-cpdef dparray dpnp_power(object x1_obj, object x2_obj, object dtype=None, dparray out=None, object where=True):
-    return call_fptr_2in_1out(DPNP_FN_POWER, x1_obj, x2_obj, dtype=dtype, out=out, where=where)
+cpdef dparray dpnp_power(utils.dpnp_descriptor x1_obj, utils.dpnp_descriptor x2_obj, object dtype=None, dparray out=None, object where=True):
```
Comment (shssf) [inline dpnp/dpnp_algo/dpnp_algo_mathematical.pyx:342]: Why we can't use the same as `call_fptr_1in_1out` (PR #775) with `call_fptr_2in_1out`?

**Comment by shssf** [comment]
It looks like this PR contains changes from several PRs

File: dpnp/dpnp_algo/dpnp_algo_mathematical.pyx [hunk 3]
```diff
@@ -339,8 +339,38 @@ cpdef dparray dpnp_negative(dpnp_descriptor x1):
...
+    if out is not None:
+        if out.dtype != result_type:
+            utils.checker_throw_value_error('power', 'out.dtype', out.dtype, result_type)
+        if out.shape != result_shape:
+            utils.checker_throw_value_error('power', 'out.shape', out.shape, result_shape)
+        result = out
+    else:
+        result = utils.create_output_array(result_shape, kernel_data.return_type, None)
+
+    """ Call FPTR function """
```
Comment (samir-nasibli) [inline dpnp/dpnp_algo/dpnp_algo_mathematical.pyx:368]:
```suggestion
    # Call FPTR function
```

File: dpnp/dpnp_algo/dpnp_algo_mathematical.pyx [hunk 4]
```diff
@@ -339,8 +339,38 @@ cpdef dparray dpnp_negative(dpnp_descriptor x1):
...
+
+    # Create result array
+    cdef dparray_shape_type x1_shape = x1_obj.shape
+    cdef dparray_shape_type x2_shape = x2_obj.shape
+    cdef dparray_shape_type result_shape = utils.get_common_shape(x1_shape, x2_shape)
+    cdef dparray result
+
+    if out is not None:
+        if out.dtype != result_type:
+            utils.checker_throw_value_error('power', 'out.dtype', out.dtype, result_type)
```
Comment (samir-nasibli) [inline dpnp/dpnp_algo/dpnp_algo_mathematical.pyx:361]: See https://github.com/IntelPython/dpnp/pull/786#discussion_r670412997

**Pull request merged**